packages/ast-spec/src/base/ClassBase.ts

```diff
@@ -0,0 +1,73 @@
+import type { Identifier } from '../expression/Identifier/spec';
+import type { ClassBody } from '../special/ClassBody/spec';
+import type { Decorator } from '../special/Decorator/spec';
+import type { TSClassImplements } from '../special/TSClassImplements/spec';
+import type { TSTypeParameterDeclaration } from '../special/TSTypeParameterDeclaration/spec';
+import type { TSTypeParameterInstantiation } from '../special/TSTypeParameterInstantiation/spec';
+import type { LeftHandSideExpression } from '../unions/LeftHandSideExpression';
+import type { BaseNode } from './BaseNode';
+
+export interface ClassBase extends BaseNode {
+  /**
+   * `true` if the class is an abstract class.
```

Comment:
Nitty style suggestion: I personally find this phrasing to be more readable. Not a big thing.

```suggestion
   * Whether the class is an abstract class.
```